common/Resources/ti.internal/extensions/util.js

```diff
@@ -0,0 +1,457 @@
+		// FIXME: Unless typeof === 'object' or 'symbol', then do util.inspect() on them
+		if (i < args.length) {
+			str += ` ${args.slice(i).map(a => String(a)).join(' ')}`;
+		}
+		return str;
+	}
+
+	// first arg wasn't string, so we loop over args and call util.inspect on each
+	return args.map(a => util.inspect(a)).join(' ');
+};
+
+/**
+ * @param {Function} constructor subclass
+ * @param {Function} superConstructor base class
+ * @returns {void}
+ */
+util.inherits = function (constructor, superConstructor) {
+	// TODO Extract common code for enforcing types of arguments. We use this logic in path.js too!
+	if (constructor == null) { // eslint-disable-line eqeqeq, no-eq-null
+		throw new TypeError(`The "constructor" argument must be of type Function. Received type ${typeof constructor}`);
+	}
+	if (superConstructor == null) { // eslint-disable-line eqeqeq, no-eq-null
+		throw new TypeError(`The "superConstructor" argument must be of type Function. Received type ${typeof superConstructor}`);
+	}
+	if (superConstructor.prototype === undefined) {
+		throw new TypeError('The "superConstructor.prototype" argument must be of type Object. Received type undefined');
+	}
+
+	constructor.super_ = superConstructor; // FIXME: Don't make enumerable!
+	Object.setPrototypeOf(constructor.prototype, superConstructor.prototype);
+};
+
+/**
+ * @param {Function} original original function to wrap which is expected to have a final callback argument
+ * @returns {Function} function that returns a Promise
+ */
+util.promisify = function (original) {
+	if (typeof original !== 'function') {
+		throw new TypeError(`The "original" argument must be of type Function. Received type ${typeof original}`);
+	}
+
+	function wrapped(...args) {
+		return new Promise((resolve, reject) => {
+			original.call(this, ...args, (err, result) => {
+				if (err) {
+					return reject(err);
+				}
+
+				return resolve(result);
+			});
+		});
+	}
+	// TODO: Copy properties from original to wrapped
+	// TODO: hook prototype chain up from wrapped to original
+	// TODO: Support custom promisify hooks
+	return wrapped;
+};
+
+/**
+ * @param {Function} original original function to convert from async/Promise return value to a callback style
+ * @returns {Function} wrapped function
+ */
+util.callbackify = function (original) {
+	if (typeof original !== 'function') {
+		throw new TypeError(`The "original" argument must be of type Function. Received type ${typeof original}`);
+	}
+
+	function wrapped(...args) {
+		const callback = args.pop();
+		const promise = original.apply(this, args);
+		promise
+			.then(result => { // eslint-disable-line promise/always-return
+				callback(null, result); // eslint-disable-line promise/no-callback-in-promise
+			})
+			.catch(err => {
+				if (!err) {
+					const wrappedError = new Error('Promise was rejected with falsy value');
+					wrappedError.reason = err;
+					err = wrappedError;
+				}
+				callback(err);
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-default;
  File: common/Resources/ti.internal/extensions/util.js;
  Line: 423;
-->

- :warning: common/Resources/ti.internal/extensions/util.js line 423 – Avoid calling back inside of a promise. (promise/no-callback-in-promise)